typing/typetexp.ml

```diff
@@ -856,7 +856,16 @@ let spellcheck ppf fold env lid =
 let fold_descr fold get_name f = fold (fun descr acc -> f (get_name descr) acc)
 let fold_simple fold4 f = fold4 (fun name _path _descr acc -> f name acc)
 
-let fold_values = fold_simple Env.fold_values
+let fold_values f =
+  (* We only use "real" values while spellchecking (as opposed to "phantom"
+     values inserted in the environment to trigger the "missing rec" hint).
+     This is needed in order to avoid dummy suggestions like:
+     "unbound value x, did you mean x?" *)
+  Env.fold_values
+    (fun name _path descr acc ->
+       match descr.val_kind with
+       | Val_unbound _ -> acc
+       | _ -> f name acc)
```

Comment:
This is an important omission of the previous patch (cc @Armael) that we need to merge independently of the rest, it could have gone into a separate commit.

Reply:
I suspect this was not detected earlier because in the case of an unbound value with a corresponding ghost binding, the "missing let rec" hint would trigger, and therefore no spellchecking would be attempted. Only the present patch adds the possibility of not printing the hint, and uses this -- buggy -- codepath.
In any case I agree this fix is required.

Reply:
I am not completely sure, but I think @Armael is right.